src/explorepy/tools.py

```diff
@@ -35,35 +36,31 @@ def get_local_time():
 
 
 def bt_scan():
-    """"Scan for bluetooth devices
-    Scans for available explore devices.
-    Prints out MAC address and name of each found device
+    """ Scan for nearby Explore devices
 
-    Args:
+    This function searches for nearby bluetooth devices and returns a list of advertising Explore devices.
 
-    Returns:
+    Note:
+        In Windows, this function returns all the paired devices and unpaired advertising devices. The 'is_paired'
+        attribute shows if the device is paired. If a device is paired, it will be in the returned list regardless if
+        it is currently advertising or not.
 
+    Returns:
+            list[namedtuple]: list of nearby devices
     """
+    NearbyDeviceInfo = namedtuple("NearbyDeviceInfo", ["name", "address", "is_paired"])
     logger.info("Searching for nearby devices...")
     explore_devices = []
     print('\n')
-    if explorepy.get_bt_interface() == 'sdk':
-        device_manager = explorepy.exploresdk.ExploreSDK_Create()
-        nearby_devices = device_manager.PerformDeviceSearch()
-        for bt_device in nearby_devices:
-            if "Explore" in bt_device.name:
-                print("Device found: %s - %s" % (bt_device.name, bt_device.address))
-                explore_devices.append((bt_device.name, bt_device.address))
-    else:
-        import bluetooth
-        nearby_devices = bluetooth.discover_devices(lookup_names=True)
-        for address, name in nearby_devices:
-            if "Explore" in name:
-                print("Device found: %s - %s" % (name, address))
-                explore_devices.append((address, name))
+    device_manager = explorepy.exploresdk.ExploreSDK_Create()
+    nearby_devices = device_manager.PerformDeviceSearch()
+    for bt_device in nearby_devices:
+        if "Explore" in bt_device.name:
+            print("Device found: %s - %s - Paired: %s" % (bt_device.name, bt_device.address, bt_device.authenticated))
+            explore_devices.append(NearbyDeviceInfo(bt_device.name, bt_device.address, bt_device.authenticated))
```

Comment:
Should we be checking if the bt_device parameters are null or already present and then gracefully exiting if they are? Potentially we don't need to, but only if you are 100% sure that it is absolutely impossible for this to go wrong

Reply:
`bt_device` definitely contains an element of `nearby_devices`. If `nearby_devices` is empty, the for loop won't iterate at all and nothing get assigned to `bt_device`.